src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockMerkleRootRule.cs

```diff
@@ -26,24 +26,22 @@ public class BlockMerkleRootRule : ConsensusRule
         /// <exception cref="ConsensusErrors.BadTransactionDuplicate">One of the leaf nodes on the merkle tree has a duplicate hash within the subtree.</exception>
         public override Task RunAsync(RuleContext context)
         {
-            if (context.CheckMerkleRoot)
-            {
-                Block block = context.BlockValidationContext.Block;
+            if (!context.CheckMerkleRoot) return Task.CompletedTask;
 
-                uint256 hashMerkleRoot2 = BlockMerkleRoot(block, out bool mutated);
-                if (block.Header.HashMerkleRoot != hashMerkleRoot2)
-                {
-                    this.Logger.LogTrace("(-)[BAD_MERKLE_ROOT]");
-                    ConsensusErrors.BadMerkleRoot.Throw();
-                }
+            var block = context.BlockValidationContext.Block;
```

Comment:
unvar please

Reply:
oh nooooo, I thought var was allowed when the type was obvious :(

Reply:
It's a bit different. 
If the type is on the right- use var. If not- explicit type.